lib/xebow/animation_frame.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Xebow.AnimationFrame do
```

Comment:
Will do.

Interesting idea for a rendering engine on top of the `Animation` abstraction. I had a thought that maybe a protocol could be used, something like:

```elixir
defprotocol Renderable do
  @spec render(animation :: %Xebow.Animation{}) :: spi_color_map()
  def render(animation)
end
```

```elixir
defimpl Renderable, for: Any do
  def render(animation) do
      animation.frame.pixel_map
      |> Enum.sort()
      |> Enum.map(fn {_cord, color} -> color end)

    data =
      Enum.reduce(colors, @sof, fn color, acc ->
        rgb = Chameleon.convert(color, Chameleon.Color.RGB)
        acc <> <<227, rgb.b, rgb.g, rgb.r>>
      end) <> @eof
  end
end
```

```elixir
defmodule Xebow.Animation do
  @derive Renderable
  defstruct [...]
end
```

I dunno though, this is pretty coupled to the fact that we're using SPI as an interface to the device?

Reply:
>this is pretty coupled to the fact that we're using SPI

Yeah, I'd lean towards decoupling. I'm not sure we've documented this yet, but one cool idea @doughsay brought up is to be able to network multiple Keybows together and render an animation across all of them (i.e. add more pixels to the RGB matrix). If animations are decoupled from the RGB matrix hopefully it will make implementing this feature easier. In other words, conversion to SPI sounds like a good responsibility of the `RGBMatrix` module.

Things like this sound like a good topic for an architecture discussion so that we can ensure the major pieces of code being refactored support the use cases envisioned for Xebow.

Reply:
Renamed `Xebow.AnimationFrame` -> `Xebow.Frame`.